### PR TITLE
Poolpi/review

### PR DESCRIPTION
**Pull request by poolpitako**

### CODE REVIEW

File: contracts/StrategyConvexCurveIronBankLP.sol [hunk 1]
```diff
@@ -49,21 +49,21 @@ contract StrategyConvexCurveIronBankLP is BaseStrategy {
     using SafeMath for uint256;
 
     ICurveFi public constant curve = ICurveFi(address(0x2dded6Da1BF5DBdF597C45fcFaa3194e53EcfeAF)); // Curve Iron Bank Pool. need this for buying more pool tokens.
-    address public crvRouter = address(0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F); // default to sushiswap, more CRV liquidity there
-    address public cvxRouter = address(0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F); // default to sushiswap, more CVX liquidity there
-    address public constant voter = address(0xF147b8125d2ef93FB6965Db97D6746952a133934); // Yearn's veCRV voter, we send some extra CRV here
```
Comment (poolpitako) [inline contracts/StrategyConvexCurveIronBankLP.sol:54]: gas saving when deploying

File: contracts/StrategyConvexCurveIronBankLP.sol [hunk 2]
```diff
@@ -49,21 +49,21 @@ contract StrategyConvexCurveIronBankLP is BaseStrategy {
...
+    address public cvxRouter = 0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F; // default to sushiswap, more CVX liquidity there
+    address public constant voter = 0xF147b8125d2ef93FB6965Db97D6746952a133934; // Yearn's veCRV voter, we send some extra CRV here
     address[] public crvPath; // path to sell CRV
     address[] public convexTokenPath; // path to sell CVX
-    
+
     address public depositContract = 0xF403C135812408BFbE8713b5A23a04b3D48AAE31; // this is the deposit contract that all pools use, aka booster
     address public rewardsContract = 0x3E03fFF82F77073cc590b656D42FceB12E4910A8; // This is unique to each curve pool, this one is for iron bank
     uint256 public pid = 29; // this is unique to each pool
-    uint256 public optimal = 0; // this is the optimal token to deposit back to our curve pool. 0 DAI, 1 USDC, 2 USDT
```
Comment (poolpitako) [inline contracts/StrategyConvexCurveIronBankLP.sol:61]: defaults are 0.

File: contracts/StrategyConvexCurveIronBankLP.sol [hunk 3]
```diff
@@ -103,24 +101,24 @@ contract StrategyConvexCurveIronBankLP is BaseStrategy {
         profitFactor = 4000; // in this strategy, profitFactor is only used for telling keep3rs when to move funds from vault to strategy (what previously was an earn call)
 
         // want = crvIB, Curve's Iron Bank pool (ycDai+ycUsdc+ycUsdt)
-        want.safeApprove(address(depositContract), uint256(-1));
+        want.safeApprove(address(depositContract), type(uint256).max);
```
Comment (poolpitako) [inline contracts/StrategyConvexCurveIronBankLP.sol:121]: `uint256(-1)` will fail in new versions of solidity.

File: contracts/StrategyConvexCurveIronBankLP.sol [hunk 4]
```diff
@@ -208,14 +204,14 @@ contract StrategyConvexCurveIronBankLP is BaseStrategy {
                 want.balanceOf(address(this))
             );
         }
-
-        return (_profit, _loss, _debtPayment);
```
Comment (poolpitako) [inline contracts/StrategyConvexCurveIronBankLP.sol:212]: when your method defines return variable with names, you don't need to return them. That's done automagically.

File: contracts/StrategyConvexCurveIronBankLP.sol [hunk 5]
```diff
@@ -304,10 +298,6 @@ contract StrategyConvexCurveIronBankLP is BaseStrategy {
 	// migrate our want token to a new strategy if needed, make sure to check claimRewards first
 	// also send over any CRV or CVX that is claimed; for migrations we definitely want to claim
     function prepareMigration(address _newStrategy) internal override {
-        uint256 stakedTokens = IConvexRewards(rewardsContract).balanceOf(address(this));
```
Comment (poolpitako) [inline contracts/StrategyConvexCurveIronBankLP.sol:307]: we discuss this privately. No withdraws in migration.

File: tests/test_change_debt_with_profit.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+
+def test_change_debt_with_profit(gov, token, vault, dudesahn, whale, strategy):
+    token.approve(vault, 2 ** 256 - 1, {"from": whale})
+    vault.deposit(100000e18, {"from": whale})
+    strategy.harvest({"from": dudesahn})
+    prev_params = vault.strategies(strategy).dict()
+
+    vault.updateStrategyDebtRatio(strategy, 25, {"from": gov})
+    token.transfer(strategy, Wei("1_000 ether"), {"from": whale})
```
Comment (poolpitako) [inline tests/test_change_debt_with_profit.py:13]: Tested airdrop and debt reduction all together

**Comment by dudesahn** [comment]
Changes incorporated, thanks for the feedback!

**Pull request closed without merging**